eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/RecoverTestAccessioningConfiguration.java

```diff
@@ -56,10 +56,12 @@ public SubmittedVariantMonotonicAccessioningService submittedVariantMonotonicAcc
             @Autowired @Qualifier("testSubmittedVariantAccessionGeneratorRecover")
             MonotonicAccessionGenerator<ISubmittedVariant> accessionGenerator,
             @Autowired SubmittedVariantAccessioningDatabaseService databaseService) {
-        return new SubmittedVariantMonotonicAccessioningService(accessionGenerator,
+        SubmittedVariantMonotonicAccessioningService sveMonotonicAccService = new SubmittedVariantMonotonicAccessioningService(accessionGenerator,
                 databaseService,
                 new SubmittedVariantSummaryFunction(),
                 new SHA1HashingFunction());
+        sveMonotonicAccService.shutDownAccessioning();
```

Comment:
I still don't really understand why some tests need the accessioning to be actually shut down, and others need the shut down to be mocked as in `RestartFailedJobTest`.  I feel like we should make sure this is relatively clear in the code, so for example if someone in the future needs to add a new test, they know which class to put it in.

Even better if we could do something to simplify - e.g. does it help to move `recoverState` out of the constructor in the accession generator? Perhaps could also help with setting the application ID as Tim mentioned...

Reply:
To answer your question, Yes it would help in certain cases, if we move the recover state out of the constructor. Some tests will be simplified like the one for testing the recovery state.
We can move recovery out of constructor, but then we need to make sure when any accessioning generator is created, the first step it does is to run the recovery (constructor is the best way to make sure it does that) or maybe we can remove the recovery state entirely from accessioning generator. We can discuss on that when we discuss the issue of putting application instance id.

1. **Shutting down service in recover tests**
These beans are created just to insert data in DB for testing recovery and we want these to release the blocks as we will be trying to recover them, hence shutting them down. As the recovery runs in constructor of the accessioning generator, we need to put the data in the DB before accessioning generator bean is created and run recovery as part of its creation. We can't put the data as part of the test's setup method, because it will be too late and all the beans would have been created by then and recovery state would have already ran, hence the workaround of creating these dummy beans to insert data.
    
3. **Mocking(Spying) the service in RestartFailedJobTest.**
In order to test the restarting of a failed job, we need to run a job and fail it and then subsequently need to run same job again by forcing it to restart. In real world, both runs of the job will be using different instances of the services as the entire stack of beans will be recreated for the restart job but that is not the case here, where the same set of beans are used for different runs. Which means we are trying to use an accession generator which is already shut down.Could not find an easier way to insert different beans for different runs hence the workaround of not shutting down the existing generator and reuse it.

Reply:
Thanks for taking the time to explain it, I think the part I wasn't getting is that these beans are mostly used to prepare the data for testing. It makes sense now and I agree we can discuss pros/cons of moving recoverState later.